### DISCUSSION TITLE
I made a simple command palette

### DISCUSSION THREAD

**Discussion by justyns** [TriliumNext/Trilium#4227] [Show and tell]
One of the things I was missing from Trilium during my initial testing was an easier way to trigger things in the mobile web ui.   The toolbar on the side just isn't big enough to have a button for everything.

I ended up making a simple command palette (similar to vscode) that can be opened with ctrl+shift+p / cmd+shift+p or swiping down on mobile.

You can find it in the comand-pallete directory here:  https://github.com/justyns/trilium-scripts

It doesn't really do much on its own, but you can add any frontend script to the list of commands by adding the attribute label `cmdPalette="Name of command"`.  I added a few shortcut notes like inserting the current timestamp since there didn't seem to be a way to do that on mobile yet.

Also fair warning, I am not a javascript or frontend developer.  I'll try to clean things up a bit, but thought I'd share the first version since it seems to work well enough for me so far.

<img width="886" alt="image" src="https://github.com/zadam/trilium/assets/215985/21f5745e-22cb-4216-94f3-21959fd8c4b2">

**Comment by meichthys** [comment]
Wow, this is so cool! I can't wait to try it out. It would be neat if we could integrate this with the default jump to menu. Perhaps we could add a hash or some other character to the beginning of the jump to search to indicate that we are issuing a command?

**Reply by zerebos** [reply]
This is actually a good idea, and much more feasible after the recent PR!

**Reply by justyns** [reply]
I like this idea too!  It'd feel more integrated at least.

 I'm not sure, but I don't think the jump to menu can be triggered in the mobile ui?   If we can add a swipe-down to trigger it, that'd be good.

In vscode, iirc the command menu is default but if you want to jump to a file you can delete `>` from the input box and it switches to a file changer.   I'd think we could do something like that or automatically put a `#` in the input box so it's easy for mobile users and desktop too

**Reply by meichthys** [reply]
^Yes this. vscode was probably the source of that idea I had 👍

**Comment by zadam** [comment]
That's pretty awesome ... command palette would be something I'd like to integrate in the near/mid future. This plugin might be a great starting point!

**Reply by justyns** [reply]
That would be awesome, let me know if you want any help with it!   Do you have any opinions on how it should be structured?   Vs relying on attributes like `cmdPalette`?

**Comment by justyns** [comment]
I added a new version of this script here: https://github.com/justyns/trilium-scripts/blob/master/command-palette/palette-testing-widget.js

It's based on the jump-to-note widget so should be more consistent with styling/etc.  I also added the ability to search for notes too.   I don't like how the notes search looks right now, but I don't think I have an easy way to get the friendly/readable version of the note path?
